### DISCUSSION TITLE
PYSCF DHF energy not matcing DIRAC DHF energy

### DISCUSSION THREAD

**Discussion by LuckyLoss** [pyscf/pyscf#2635] [Q&A] [answered]
I have been attempting to run some 4 component DHF calculations in Pyscf:

I set up the calculation of h2o in the 3-21g basis and found the DHF SCF Energy: -75.6270551569308 Hartree

I tried setting up the same calculation (same basis, coordinates, etc) in the DIRAC code (https://www.diracprogram.org) and  found DHF SCF Energy:  -75.634439758094146 Hartree. I find this reasonable.

Similarly I set up a set a calculation of O in the aug-ccpvdz basis. I found DHF SCF Energy: -74.82065483109689 Hartree

However, setting up the same calculation in DIRAC code I find DHF SCF Energy: -68.036423909727844 Hatree

This difference is rather surprising and I am not sure if it is reasonable (and this is just one example of a discrepancy but there are more) . I looked into the pyscf DHF source code and found the following comment: 
'''
    Visscher point charge corrections for small component, TCA, 98, 68
    Note there is a small difference to Visscher's work. The model
    charges in Visscher's work are obtained from atomic calculations.
    Charges here are Mulliken charges on small components.
    '''

I know the DIRAC code is developed by Visscher and this is the only difference I could find between the codes.  Is this the cause of the discrepancy or is it something else? Which assumption is better? Is this a bug? I would appreciate guidance in interpreting the results. Furthermore is this is not a bug, I would appreciate any argument (other than comparing to experiment) why one code might be closer to the true state of the system than the other.

**Comment by sunqm** [comment, marked as answer]
Three different treatments between pyscf and Dirac.
* the finite-size nuclear model (Dirac) vs point nuclear model (pyscf) by default.
* the unrestricted kinetic balance (Dirac) vs restricted kinetic balance (pyscf).
* light speed may be different, at 5th or 6th decimal place.

However, none of them are likely the source of difference. The HF energy -68.036423909727844 does not look reasonable for light-element molecules like water.  You can try to change light speed in Dirac, to a larger number, to verify whether the HF energy converges to the non-relativistic limit.

**Comment by LuckyLoss** [comment]
Thank you for you response; I took  the non-relativistic limit as you suggested and the answers for Oxygen match at around -74.8.
 
1) As a follow-up, could you clarify why PySCF DHF approach captures this energy, whereas for DIRAC DHF, it was necessary to take the non-relativistic limit? I would like to understand why this discrepancy occurred in the first place.

2 ) Additionally, is there anyway to setting in PySCF DHF to achieve the same result as **relativistic** DIRAC result?

**Comment by sunqm** [comment]
The results for the oxygen atom provided by DIRAC seems incorrect. You may need to consult the DIRAC developers to identify the cause of the errors.
